packages/compat/src/prepare-htmlbars-ast-plugins.ts

```diff
@@ -0,0 +1,28 @@
+import { join } from 'path';
+
+export default function prepHtmlbarsAstPluginsForUnwrap(registry: any): void {
```

Comment:
should this be unit tested to prevent future regressions?

Reply:
I would suggest setting `THROW_UNLESS_PARALLELIZABLE` during something like `stage1.test.ts`.

We already run with this flag, but only in the macros package's test suite.

Reply:
Setting `THROW_UNLESS_PARALLELIZABLE=1` in `stage1.test.ts` definitely causes the tests to fail. I was able to confirm that the patch here works. However, even with this patch, the tests continue to fail as the `inject-babel-helpers` plugin (from `ember-source`) is not parallelizable.

This was fixed in https://github.com/emberjs/ember.js/pull/19047, but not backported to `ember-source@3.17` which is currently used in a few test-packages (e.g. [here](https://github.com/embroider-build/embroider/blob/153a074d59ebacb6ff9b867c3376e61c0992946f/test-packages/support/package.json#L21) and [here](https://github.com/embroider-build/embroider/blob/a122d4d356c3ad185745c21829be78e13b59d1bf/test-packages/fastboot-addon/package.json#L47)).

Bumping `ember-source` resolves the issue, however, I don't think I can currently bump these instances of `ember-source` above `3.17` per https://github.com/embroider-build/embroider/issues/927 / https://github.com/embroider-build/embroider/pull/934 (@stefanpenner).

Reply:
Ok that makes sense. @eoneill how about:

1) we merge this
2) open an issue specifically to enable `THROW_UNLESS_PARALLELIZABLE=1` once ember-source is bumped across the project

Reply:
@eoneill some of the ember-sources' are at the right version, can additional `THROW_UNLESS_PARALLELIZABLE` be turned on, or are we blocked from completely until ember-source is upgraded.

Also, what is the lowest version of ember-source that satisfies the requirement?

Reply:
> Bumping ember-source resolves the issue, however, I don't think I can currently bump these instances of ember-source above 3.17 per #927 / #934 (@stefanpenner)

@eoneill  #927 / #934 are not ember-source, they are for ember-cli, but the fix appears to be in ember-source -> https://github.com/emberjs/ember.js/pull/19047/commits/104265a63c8dd4e9f28972223be7ddfbb914381f

Looking at the ember-source commit, it suggests that if we get to ember-source@~3.22 the issue you describe is addressed.

```
rg 'ember-source":'
test-packages/support/package.json
21:    "ember-source": "3.17.0",

packages/router/package.json
63:    "ember-source": "~3.16.0",

test-packages/macro-sample-addon/package.json
54:    "ember-source": "~3.10.0",

test-packages/fastboot-addon/package.json
47:    "ember-source": "~3.17.0",

test-packages/sample-transforms/package.json
46:    "ember-source": "~3.10.0",

packages/util/package.json
64:    "ember-source": "~3.21.1",

tests/app-template/package.json
54:    "ember-source": "~3.26.1",

tests/addon-template/package.json
53:    "ember-source": "~3.26.1",
```

I can quickly try to upgrade those, and see if we can land this with the flag enabled

Reply:
Let's see if we can just quickly sneak in a conservative ember-source bump to help with https://github.com/embroider-build/embroider/pull/935